src/main/java/org/fogbowcloud/manager/core/models/linkedlist/SynchronizedDoubleLinkedList.java

```diff
@@ -0,0 +1,108 @@
+package org.fogbowcloud.manager.core.models.linkedlist;
+
+import org.fogbowcloud.manager.core.models.orders.Order;
+
+public class SynchronizedDoubleLinkedList implements ChainedList {
+
+	private Node head;
+	private Node tail;
+	private Node current;
+
+	public SynchronizedDoubleLinkedList() {
+		this.head = this.tail = this.current = null;
+	}
+
+	public synchronized Node getCurrent() {
+		return this.current;
+	}
+
+	protected synchronized Node getHead() {
+		return this.head;
+	}
+
+	protected synchronized Node getTail() {
+		return this.tail;
+	}
+
+	@Override
+	public synchronized void addItem(Order order) {
+		if (order == null) {
+			throw new IllegalArgumentException("Order cannot be null.");
+		}
+		if (this.head == null) {
+			Node firstNode = new Node(null, order, null);
+			this.tail = this.head = this.current = firstNode;
+		} else {
+			Node newItem = new Node(this.tail, order, null);
+			this.tail.setNext(newItem);
+			this.tail = newItem;
+		}
+	}
```

Comment:
We need to check if current was in the end of the list (current == null). In this case, current should point to the newItem just added. Simply add:

    if (this.current == null) this.current = this.tail;

Add a comment to explain this, as it is done in the removeItem() method.